### DISCUSSION TITLE
[Feature] External Library Metadata Filter

### DISCUSSION THREAD

**Discussion by DomDomonom** [immich-app/immich#14646] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Hey!
Love this software, it's so impressive! The team that contribute to it are badass.
For my workflow as a professional photographer, I want to be able to put a metadata filter on external libraries to determine what is pulled into immich, much like the existing file name exclusions. 
Specifically I want to be able to tell immich to only pull images that have 4 stars rating or higher, and ignore all other images. I'm sure there's plenty of other people who would have other metadata exclusions that would be useful to them.
Ratings are incredibly important for my workflow, and there's a lot of images that I deliver to clients that are more or less chaff to me personally. I know @stumpigit is working on adding a rating search filter that will make immich much more useful for me already, but ignoring those images altogether and not having them fill up my library would be very useful for me. 
I imagine this would have a significant performance cost on external library syncing, but to me it would be worth it as it would reduce further in-immich processing and library size.
I don't know if immich is intended for professional photographers to manage an archive, but I know a lot of my colleagues who find dealing with their large archives frustrating, and I would love to recommend immich to everyone!

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
This would complicate the library import logic significantly, so we have no plans to add it. Maybe #1673 will cover some of what you need.

**Reply by DomDomonom** [reply]
Thanks for the reply! That's fair and yes that would also go a long way, thanks for sharing!

**Comment by jo-gross** [comment]
@DomDomonom i also have the same issue. Do you or your collegues have found a work-around? 
Mine is to make a copy of the filtered photos to a seperate folder and only include those, but i am not very happy with this approach.
